src/Libraries/Microsoft.Extensions.Resilience/Resilience/ResilienceContextExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Extensions.Http.Telemetry;
+using Microsoft.Shared.Diagnostics;
+using Polly;
+
+namespace Microsoft.Extensions.Resilience.Resilience;
+
+/// <summary>
+/// Extensions for <see cref="ResilienceContext"/>.
+/// </summary>
+public static class ResilienceContextExtensions
+{
+    private static readonly ResiliencePropertyKey<RequestMetadata?> _requestMetadataKey = new(TelemetryConstants.RequestMetadataKey);
+
+    /// <summary>
+    /// Sets the <see cref="RequestMetadata"/> to the <see cref="ResilienceContext"/>.
+    /// </summary>
+    /// <param name="context">The context instance.</param>
+    /// <param name="requestMetadata">The request metadata.</param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="context"/> or <paramref name="requestMetadata"/> is <see langword="null"/>.</exception>
```

Comment:
```suggestion
    /// <exception cref="ArgumentNullException"><paramref name="context"/> or <paramref name="requestMetadata"/> is <see langword="null"/>.</exception>
```

@gewarren is this ok? Or should it be changed to this?
```suggestion
    /// <exception cref="ArgumentNullException">
    ///   <paramref name="context"/> is <see langword="null"/>
    ///   <p>- or -</p>
    ///   <paramref name="requestMetadata"/> is <see langword="null"/>.
    ///</exception>
```

Reply:
@RussKie I think your first suggestion is fine. (As an aside, I also think you need to use `<para>` instead of `<p>`.)

Reply:
Actually, you don't need <p> or <para>. Just a blank line is sufficient.